### PR TITLE
Fix `qml.draw/_mpl` to work with deferred measurements circuits that use MCMs in measurements

### PR DESCRIPTION
**Pull request by mudit2812**
**Context:**
`defer_measurements` leaves `MeasurementValue`s in the terminal measurement processes if they are present. This is not friendly with the drawer as the classical wires for those measurement values do not exist when using `defer_measurements`. This PR adds a "hack" fix so that the drawer/MPL drawer work with this scenario.

**Description of the Change:**
Add `transform_deferred_measurements_tape` util function, which replaces any `MeasurementValue`s present in the tape after applying `defer_measurements` with wires. This transform is only applied in `tape_text` and `tape_mpl`.

**Benefits:**
Drawer works better with MCMs

**Possible Drawbacks:**
Hacky fix, technical debt to implement cleaner fix later

**Related GitHub Issues:**
https://github.com/PennyLaneAI/pennylane/issues/5588

### CODE REVIEW

File: doc/releases/changelog-0.36.0.md [hunk 1]
```diff
@@ -551,6 +551,10 @@
 
 <h3>Bug fixes 🐛</h3>
 
+* Fixed a bug where `qml.draw` and `qml.draw_mpl` incorrectly raised errors for circuits collecting statistics on mid-circuit measurements
+  while using `qml.defer_measurements`.
+  [(#)]()
```
Comment (mudit2812) [inline doc/releases/changelog-0.36.0.md:556]:
```suggestion
  [(#5610)](https://github.com/PennyLaneAI/pennylane/pull/5610)
```

**Comment by mudit2812** [comment]
[sc-62204]

File: tests/transforms/test_defer_measurements.py [hunk 2]
```diff
@@ -1580,6 +1580,56 @@ def qfunc():
...
+    @pytest.mark.parametrize("mp", [qml.sample, qml.probs, qml.var, qml.counts, qml.expval])
+    def test_draw_mpl_with_mcm_terminal_measure(self, mp):
+        """Test that no error is raised when drawing a circuit which collects
+        statistics on mid-circuit measurements"""
+
+        def qfunc():
+            m_0 = qml.measure(0, reset=True)
+            qml.cond(m_0, qml.RY)(0.312, wires=1)
+
+            return mp(op=m_0)
```
Comment (dwierichs) [inline tests/transforms/test_defer_measurements.py:1625]: To cover the uncovered test line, can do
```suggestion
            return mp(op=m_0), qml.expval(qml.Z(1))
```

**Review by dwierichs — CHANGES_REQUESTED** [review]
Nice, thanks for the quick fix @mudit2812 :)
Just one suggestion to cover the uncovered line within a test you added.

**Review by dwierichs — APPROVED** [review]
Thanks!

**Pull request merged**